docs/20_smart-contracts/40_guides/10_resources_guide.md

```diff
@@ -0,0 +1,232 @@
+
+* [The NET configuration variables](https://github.com/AntelopeIO/leap/blob/a4c29608472dd195d36d732052784aadc3a779cb/libraries/chain/include/eosio/chain/config.hpp#L57)
+* [The transaction initialization](https://github.com/AntelopeIO/leap/blob/e55669c42dfe4ac112e3072186f3a449936c0c61/libraries/chain/controller.cpp#L1559)
+* [The transaction NET billing](https://github.com/AntelopeIO/leap/blob/e55669c42dfe4ac112e3072186f3a449936c0c61/libraries/chain/controller.cpp#L1577)
+* [The check of NET usage for a transaction](https://github.com/AntelopeIO/leap/blob/a4c29608472dd195d36d732052784aadc3a779cb/libraries/chain/transaction_context.cpp#L376)
+
+### How To Rent NET
+
+For details on how to rent NET resources refer to the how to [Account Power Up](#account-power-up) section.
+
+## Resource Cost Estimation
+
+As a developer if you want to estimate how much CPU and NET is required for a transaction to be executed you can employ one of the following methods:
+
+* Use the `--dry-run` option for the `dune -- cleos push transaction` command.
+* Use any tool that can pack a transaction and send it to the blockchain and specify the `--dry-run` option.
+* Use the chain API endpoint [`compute_transaction`](https://github.com/AntelopeIO/leap/blob/51c11175e54831474a89a449beea1fb067e3d1e9/plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp#L489).
+
+In all cases, when the transaction is processed, the blockchain node simulates the execution of the transaction and, as a consequence, the state of the blockchain is changed speculatively, allowing for the CPU and NET measurements to be done. However in the end the transaction is not sent to the blockchain and the caller receives in return the estimated CPU and NET costs.
+
+## Account Power Up
+
+To power up an account means to rent CPU and NET from the PowerUp resource model which is implemented as a smart contract on the blockchain and allocate them to the account of your choice. The action to power up an account is `powerup`. It takes as parameters:
+
+* The `payer` of the fee, must be a valid EOS account.
+* The `receiver` of the resources, must be a valid EOS account.
+* The `days` which must always match `state.powerup_days` specified in the [PowerUp configuration settings](https://github.com/eosnetworkfoundation/eos-system-contracts/blob/7cec470b17bd53b8c78465d4cbd889dbaf1baffb/contracts/eosio.system/include/eosio.system/eosio.system.hpp#L588).
+* The `net_frac`, and the `cpu_frac` are the percentage of the resources that you need. The easiest way to calculate the percentage is to multiple 10^15 (100%) by the desired percentage. For example: 10^15 * 0.01 = 10^13.
```

Comment:
Is there really no easier way than doing these calculations yourself? 

This is really not easy for devs to understand.

Reply:
I know, it is not simple... I will research more and see if I can find something else

Reply:
I've create this ticket/work item to track this matter
https://github.com/orgs/eosnetworkfoundation/projects/18/views/10?filterQuery=vertical%3ADocs+-status%3ADone&pane=issue&itemId=20399343